### PR TITLE
Drask Traitor Item - Cryoregenerative Enhancer

### PR DESCRIPTION
**Pull request by PollardTheDragon**
## What Does This PR Do
This PR adds the Cryoregenerative Enhancer - a Drask species-specific traitor item.

It costs 25 TC, and gives the purchaser a nanite injector that applies the TRAIT_DRASK_SUPERCOOL trait. The nanite injector only works on Drask - it also does not work on changelings or vampires. While you have this trait, it quadruples the amount of healing you get per breathe of cold air - whether it be from internals or the room. If your lungs are replaced with non-drask lungs while you have this trait, it will work against you by increasing the amount of cold damage you take from air by the same amount. Once injected, you're left with an empty injector.

The healing follows the three tiers of cold air:
1: 1 brute and 0.5 burn (Breath Temp < 260K)
2: 3 brute and 1.5 burn (Breath Temp <200K)
3: 4.5 Brute and 2.25 burn (Breath Temp <120K)

This item should be S-Class.

## Why It's Good For The Game
Drask currently do not have a traitor item. This PR adds an item for Drask traitors that enhances their species trait while not being overwhelming. 

## Images of changes
https://github.com/ParadiseSS13/Paradise/assets/144391971/d8b680e6-6ea1-4aa1-9b01-b0b7535e594d

## Testing
Loaded in as drask. Supercooled a tank of oxygen. Beat self with syndie knuckledusters. Inhaled super cold air. Healed.

## Changelog
:cl:
add: Cryoregenerative Enhancer - Drask Traitor Item
/:cl:

### CODE REVIEW

File: code/game/objects/items/devices/traitordevices.dm [hunk 1]
```diff
@@ -286,6 +286,39 @@
...
+/obj/item/cryoregenerative_enhancer
+	name = "cryoregenerative enhancer"
+	desc = "Specially designed nanomachines that enhance the low-temperature regenerative capabilities of drask. Requires supercooled air in the enviroment or internals to function."
+	icon = 'icons/obj/hypo.dmi'
+	icon_state = "combat_hypo"
+	var/used = FALSE
+
+/obj/item/cryoregenerative_enhancer/examine_more(mob/user)
+	. = ..()
+	. = list()
```
Comment (Fordoxia) [inline code/game/objects/items/devices/traitordevices.dm:298]:
```suggestion
```
Not needed unless you have a subtype of the injector with a different description.

**Review by Fordoxia — APPROVED** [review]
Code looks good to drask.

File: code/modules/surgery/organs/organ_datums/lung_datum.dm [hunk 2]
```diff
@@ -249,6 +249,8 @@
 			TC = cold_level_1_damage
 		if(TC)
 			for(var/D in cold_damage_types)
+				if(HAS_TRAIT(H, TRAIT_DRASK_SUPERCOOL))
+					TC = TC * 4
```
Comment (Sirryan2002) [inline code/modules/surgery/organs/organ_datums/lung_datum.dm:253]:
```suggestion
					TC *= 4
```

File: code/datums/uplink_items/uplink_traitor.dm [hunk 3]
```diff
@@ -396,6 +396,17 @@
 	cost = 25 /// A fresh start, but a start with nothing. Hard to use as well
 	species = list("Grey")
 
+//Drask
```
Comment (1080pCat) [inline code/datums/uplink_items/uplink_traitor.dm:399]:
```suggestion
// Drask
```

File: code/game/objects/items/devices/traitordevices.dm [hunk 4]
```diff
@@ -286,6 +286,38 @@
...
+		return
+	if(used)
+		to_chat(user, "<span class='notice'>The injector is empty!</span>")
+		return
+	used = TRUE // Set this BEFORE the popup to prevent people using the injector more than once.
+	var/choice = tgui_alert(user, "The injector is still unused. Do you wish to use it?", "Cryoregenerative enhancer", list("Yes", "No"))
+	if(choice != "Yes")
+		to_chat(user, "<span class='notice'>You decide against using [src].</span>")
+		used = FALSE
+		return
```
Comment (Sirryan2002) [inline code/game/objects/items/devices/traitordevices.dm:317]:
```suggestion
	if(used)
		to_chat(user, "<span class='notice'>The injector is empty!</span>")
		return
	var/choice = tgui_alert(user, "The injector is still unused. Do you wish to use it?", "Cryoregenerative enhancer", list("Yes", "No"))
	if(choice != "Yes")
		to_chat(user, "<span class='notice'>You decide against using [src].</span>")
		return
	if(used)
		to_chat(user, "<span class='notice'>The injector is empty!</span>")
		return
	used = TRUE 
```
doing the checks like this is a bit more intuitive for future readers

**Review by Sirryan2002 — APPROVED** [review]

File: code/game/objects/items/devices/traitordevices.dm [hunk 5]
```diff
@@ -286,6 +286,38 @@
...
+	. = ..()
+	. += "Designed by Viim-vaarooomunnm's prestigious polytechnic university, these experimental nanomachines infiltrate the cells of the drask host and integrate into the specialised cryoregenerative organelles that facilitate low-temperature healing and work to boost enzymatic activity, massively improving the efficiency of the associated metabolic processes."
+	. += ""
+	. += "Clinical trials have shown a four times increase in the rate of healing compared to a placebo. Whilst the product is technically not yet available to the public, your connections have allowed you to get a hold of them early..."
+
+/obj/item/cryoregenerative_enhancer/attack_self(mob/living/user)
+	if(HAS_TRAIT(user, TRAIT_DRASK_SUPERCOOL))
+		to_chat(user, "<span class='warning'>Your regeneration is already enhanced!</span>")
+		return
+	if(user.mind && (IS_CHANGELING(user) || user.mind.has_antag_datum(/datum/antagonist/vampire)) || (user.dna && user.dna.species.name != "Drask"))
```
Comment (DGamerL) [inline code/game/objects/items/devices/traitordevices.dm:306]:
```suggestion
	if(user.mind && (IS_CHANGELING(user) || user.mind.has_antag_datum(/datum/antagonist/vampire)) || user.dna?.species.name != "Drask")
```

**Review by DGamerL — APPROVED** [review]

File: code/game/objects/items/devices/traitordevices.dm [hunk 6]
```diff
@@ -286,6 +286,38 @@
...
+	desc = "Specially designed nanomachines that enhance the low-temperature regenerative capabilities of drask. Requires supercooled air in the enviroment or internals to function."
+	icon = 'icons/obj/hypo.dmi'
+	icon_state = "combat_hypo"
+	var/used = FALSE
+
+/obj/item/cryoregenerative_enhancer/examine_more(mob/user)
+	. = ..()
+	. += "Designed by Viim-vaarooomunnm's prestigious polytechnic university, these experimental nanomachines infiltrate the cells of the drask host and integrate into the specialised cryoregenerative organelles that facilitate low-temperature healing and work to boost enzymatic activity, massively improving the efficiency of the associated metabolic processes."
+	. += ""
+	. += "Clinical trials have shown a four times increase in the rate of healing compared to a placebo. Whilst the product is technically not yet available to the public, your connections have allowed you to get a hold of them early..."
```
Comment (Fordoxia) [inline code/game/objects/items/devices/traitordevices.dm:300]:
```suggestion
	. += "Clinical trials have shown a four times increase in the rate of healing compared to a placebo. Whilst the product is technically not yet available to the public, the right connections with the right people allow interested parties to obtain samples early..."
```
I just realised that someone else may examine the item. It'd be pretty weird for the HoS to be told about their syndicate connections as they examine the item.

File: code/game/objects/items/devices/traitordevices.dm [hunk 7]
```diff
@@ -309,12 +309,14 @@
...
 		to_chat(user, "<span class='notice'>The injector is empty!</span>")
 		return
-	used = TRUE // Set this BEFORE the popup to prevent people using the injector more than once.
 	var/choice = tgui_alert(user, "The injector is still unused. Do you wish to use it?", "Cryoregenerative enhancer", list("Yes", "No"))
 	if(choice != "Yes")
 		to_chat(user, "<span class='notice'>You decide against using [src].</span>")
-		used = FALSE
 		return
+	if(used)
+		to_chat(user, "<span class='notice'>The injector is empty!</span>")
```
Comment (Fordoxia) [inline code/game/objects/items/devices/traitordevices.dm:317]:
```suggestion
		to_chat(user, "<span class='warning'>The injector is empty!</span>")
```

**Review by lewcc — APPROVED** [review]

**Pull request merged**